examples/sample_measurement/measurement.py

```diff
@@ -4,6 +4,7 @@
 from enum import Enum
 
 import click
+from examples.sample_measurement.assets.color_pb2 import ProtobufColor
```

Comment:
Hmm, I'm not sure how to best resolve this without trying to separate the acceptance tests from using sample_measurement service. 

I thought we could include the assets in the examples.zip and then reference `assets.color_pb2` but I don't think that works if we don't make assets an module because the tests break. Perhaps we can create a module that contains just `colors.proto` and associated files that sample_measurement and tests can use, but that seems a bit excessive. We could remove the protobuf enum from sample_measurement and implement test coverage at a lower level as an alternative.

Note that importing `examples.sample_measurement.assets.color_pb2` does work when running the example directly and with running tests on my machine, but I suppose that's only because I'm running it out of the repo where the assets files can be found in the examples directory.

Reply:
Let's do a conditional import for now and consider separating the examples from the tests later.

```
try:
    from .assets import color_pb2
except ImportError:
    from examples.sample_measurement.assets import color_pb2
```

Reply:
Switched to using a conditional import as suggested.

Reply:
@bkeryan I'm hitting some mypy issues when it runs on this PR and I'm not sure how best to resolve them. 

These do not appear when I run locally via `poetry run mypy -p examples`. I tried directly specifying `sample_measurement` via `poetry run mypy .\examples\sample_measurement\` (which is how it looks like the PR checks are running this check). I do get a "Source file found twice under different module names", but I get none of the same errors from this PR's example check. I'm not sure mypy that's being run on this PR likes the conditional import workaround or respecting the `TYPE_CHECK` condition to import `EnumTypeWrapper`. 

Doing some investigating on stackoverflow, I suspect that the PR's mypy is configured differently than what I have locally, but I'm unsure fighting with mypy into ignoring these errors is the right approach. I tried making sample_measurement's a module by adding a `__init__.py` for that directory as well and changing the imports to reference differently without much success. Please let me know if you have any ideas on how to get the examples check passing.

Reply:
These problems are why the GitHub "check examples" works the way it does: https://github.com/ni/measurementlink-python/blob/main/.github/workflows/check_examples.yml

You need to run `poetry install` and `poetry run mypy .` separately for each example.

Reply:
https://github.com/ni/measurementlink-python/actions/runs/5557275444/jobs/10150844869?pr=328

```
  measurement.py:11: error: Cannot find implementation or library stub for module named "examples.sample_measurement._stubs"  [import]
  measurement.py:11: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
  measurement.py:11: error: Name "color_pb2" already defined (by an import)  [no-redef]
```
The first error can be avoided by setting `[[tool.mypy.overrides]]` appropriately: https://github.com/ni/measurementlink-python/blob/main/examples/nidmm_measurement/pyproject.toml#L25

But the second error probably needs `# type: ignore`. 

I think I saw the no-redef error here too but never got around to posting a PR for the fix: https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/__init__.py

```
  measurement.py:[50](https://github.com/ni/measurementlink-python/actions/runs/5557275444/jobs/10150844869?pr=328#step:7:59): error: Argument "enum_type" to "configuration" of "MeasurementService" has incompatible type "type[ProtobufColor]"; expected "Optional[type[Enum]]"  [arg-type]
  measurement.py:61: error: Argument "enum_type" to "output" of "MeasurementService" has incompatible type "type[ProtobufColor]"; expected "Optional[type[Enum]]"  [arg-type]
  Found 4 errors in 1 file (checked 4 source files)
```

This is because the example is using the released `ni-measurementlink-service` 1.1.0 which does not include your changes. You need to uncomment the ../.. stuff again.